ext/ffi/jit_trampoline.rs

```diff
@@ -32,7 +32,8 @@ fn native_arg_to_c(ty: &NativeType) -> &'static str {
     NativeType::I64 => "int64_t",
     NativeType::ISize => "intptr_t",
     NativeType::USize => "uintptr_t",
-    NativeType::Pointer | NativeType::Function => "void*",
+    NativeType::Pointer => "struct FastApiTypedArray*",
+    NativeType::Function => "void*",
```

Comment:
Should we change this to use the `void(*pN)()` type while we're here?

Reply:
Please elaborate why it should be such type

Reply:
Because there's no guarantee on a language level that a data pointer is valid as a function pointer.

It is so for all the envs that Deno supports, but technically they may not be the same.

Reply:
@littledivy please take a look

Reply:
Admittedly this is very rare: Embedded systems using the Harvard architecture have smaller function pointer size. Some others might also be like this but it's really not a big consideration.

Reply:
We already have a const assert for this assumption